### PR TITLE
Music System Improvements

### LINKED ISSUE #169: Multiple Lavalink servers
To solve audio issues when the bot is used in regions other than `us-east`, we could add a system that selects the nearest lavalink server.

### LINKED ISSUE #186: s!volume command responds wrongly whenever not playing anything
**Bug description**
s!volume command makes the bot respond with "I ain't playing anything!" whenever the bot itself is not in a voice channel

**How to reproduce**
Steps to reproduce the unusual behavior:
1. Make sure the bot is not in a voice channel
2. Run s!volume 99

**Expected behavior**
It should respond with something among the lines of "Can't adjust volume while not playing anything"

**Screenshots**
https://puu.sh/AP3rG/060adf2476.png

**Additional context**
None

### LINKED ISSUE #392: Add ability to remove a song from queue

### LINKED ISSUE #453: Earrape command
```js
let bands = [];
for (let i = 0; i < 6; i++) {
    bands.push({band: i, gain: 1});
}
const player = this.client.playerManager.get(message.guild.id);
player.setEQ(bands);
```

and then max out the volume

### LINKED ISSUE #456: Add a "skip to" command
`s!skipto<song index>` would skip to a specific song on the playlist, removing everything before it.

### LINKED ISSUE #457: Add a flag to s!play to search on SoundCloud instead of youtube
`s!play darude sandstorm --soundcloud` (maybe `--sc` too!)

### LINKED ISSUE #459: Make s!genius search lyrics for the song you're listening to if no arguments are provided
Maybe if we could make it work with spotify, too. Would be awesome.

### LINKED ISSUE #461: Add a seek command to let users jump time in a song
`s!seek 10` - Jumps 10 seconds
`s!seek -10` - Jumps 10 seconds **backwards**
`s!seek 1:25` - Jumps **to** 1'25" seconds on the current song

### LINKED ISSUE #462: Add a shuffle command to let people shuffle the current playlist

### LINKED ISSUE #481: Bot won't let you queue if it's in a channel it couldn't normally join
**How to reproduce**
Steps to reproduce the unusual behavior:
1. Go to a channel the bot **can** join
2. Start playing something with `s!play`
3. Move the bot to a channel it can't join normally
4. Try to queue something with `s!play`

**Expected behavior**
Bot should queue the song normally, as it's already playing something else

**Screenshots**
image

### LINKED ISSUE #485: Add Deezer support to s!play
Just like we support spotify

### LINKED ISSUE #552: s!clear
**Is your feature request related to a problem? Please describe.**
after adding a playlist it's hard to skipp the songs when we get bored of it.

**Describe the solution you'd like**
add votes to s!stop or add s!clear that would empty the queue except the one that's playing

**Describe alternatives you've considered**
ban doges

**Additional context**
image

### LINKED ISSUE #592: Problems on playong soundcloud music
Error screenshot
# Bug description 
Error on displaying the track cover, leading to an erro on the music playback
# How reproduce
ps. the screenshot isnt mine so i cant tell details
Play a soundcloud music
# Expected behaviour
Displaying the track cover and playing the music fisplaying no errors

### PR DESCRIPTION
**Pull request by davipatury**
### Fixes
- [x] Fixes #186 
- [x] Fixes #481 
- [x] Fixes #592
### Features
- [x] Closes #169 
- [x] Closes #457 
- [x] Closes #485 (?)
- [x] Closes #459
### Commands
- [x] Closes #552 (`s!queue clear`)
- [x] Closes #462 (`s!queue shuffle`)
- [x] Closes #461 (`s!seek`)
- [x] Closes #456 (`s!skipto`?)
- [x] Closes #453 (`s!bassboost`?)
- [x] Closes #392 (`s!queue remove <index>`)

### THIS PULL REQUEST CHANGES SOME ENVIRONMENT VARIABLES

### CODE REVIEW

**Comment by pedrofracassi** [comment]
yo, while we're on this: queue pagination, maybe?

File: src/commands/music/queue.js [hunk 1]
```diff
@@ -9,26 +10,146 @@ module.exports = class Queue extends Command {
...
+      new QueueRemove(client, this),
+      new QueueShuffle(client, this)
+    ]
+
     this.requirements = new CommandRequirements(this, { guildOnly: true, guildPlaying: true })
   }
 
-  async run ({ t, author, channel, guild }) {
-    const guildPlayer = this.client.playerManager.get(guild.id)
+  async run ({ t, aliase, author, channel, guild, prefix }) {
```
Comment (pedrofracassi) [inline src/commands/music/queue.js]: Hey, just a heads up: [The singular form of **aliases** is **alias**, not **aliases**](https://www.dictionary.com/browse/aliases).

File: src/locales/en-US/commands.json [hunk 2]
```diff
@@ -144,7 +144,28 @@
     "noTrackIdentifier": "You need to give me a track name or URL!"
   },
   "queue": {
-    "commandDescription": "Shows the current queue"
+    "commandDescription": "Shows the current queue",
+    "clearAdvice": "You can clear the music queue using `{{usage}}`.",
+    "subcommands": {
+      "clear": {
+        "commandDescription": "Clears the current queue",
+        "queueCleared": "Music queue has been cleared!"
```
Comment (pedrofracassi) [inline src/locales/en-US/commands.json]:
```suggestion
        "queueCleared": "The queue has been cleared!"
```

File: src/locales/en-US/commands.json [hunk 3]
```diff
@@ -144,7 +144,28 @@
...
+      },
+      "shuffle": {
+        "commandDescription": "Shuffles the current queue",
+        "queueShuffled": "Music queue has been shuffled!"
```
Comment (pedrofracassi) [inline src/locales/en-US/commands.json]:
```suggestion
        "queueShuffled": "The queue has been shuffled!"
```

File: src/locales/en-US/commands.json [hunk 4]
```diff
@@ -144,7 +144,28 @@
...
+      },
+      "remove": {
+        "commandDescription": "Removes a song from the current queue",
```
Comment (pedrofracassi) [inline src/locales/en-US/commands.json]:
```suggestion
        "commandDescription": "Removes a song from the queue",
```

File: src/locales/en-US/commands.json [hunk 5]
```diff
@@ -144,7 +144,28 @@
...
+        "missingIndexParameter": "You need to give me a song index to remove from the queue",
+        "songRemoved": "{{- songName}} has been removed from the queue"
+      },
+      "jump": {
+        "commandDescription": "Removes a song from the current queue",
+        "missingIndexParameter": "You need to give me a song index to jump to",
```
Comment (pedrofracassi) [inline src/locales/en-US/commands.json]:
```suggestion
        "missingIndexParameter": "You need to give me the number of the song you want to jump to",
```

File: src/locales/en-US/commands.json [hunk 6]
```diff
@@ -144,7 +144,28 @@
...
+        "commandDescription": "Clears the current queue",
+        "queueCleared": "Music queue has been cleared!"
+      },
+      "shuffle": {
+        "commandDescription": "Shuffles the current queue",
+        "queueShuffled": "Music queue has been shuffled!"
+      },
+      "remove": {
+        "commandDescription": "Removes a song from the current queue",
+        "missingIndexParameter": "You need to give me a song index to remove from the queue",
```
Comment (pedrofracassi) [inline src/locales/en-US/commands.json]:
```suggestion
        "missingIndexParameter": "You need to give me the number of the song you want to remove from the queue",
```

File: src/locales/en-US/commands.json [hunk 7]
```diff
@@ -144,7 +144,28 @@
     "noTrackIdentifier": "You need to give me a track name or URL!"
   },
   "queue": {
-    "commandDescription": "Shows the current queue"
+    "commandDescription": "Shows the current queue",
+    "clearAdvice": "You can clear the music queue using `{{usage}}`.",
+    "subcommands": {
+      "clear": {
+        "commandDescription": "Clears the current queue",
```
Comment (pedrofracassi) [inline src/locales/en-US/commands.json]:
```suggestion
        "commandDescription": "Clears the queue",
```

File: src/locales/en-US/commands.json [hunk 8]
```diff
@@ -144,7 +144,28 @@
     "noTrackIdentifier": "You need to give me a track name or URL!"
   },
   "queue": {
-    "commandDescription": "Shows the current queue"
+    "commandDescription": "Shows the current queue",
+    "clearAdvice": "You can clear the music queue using `{{usage}}`.",
```
Comment (pedrofracassi) [inline src/locales/en-US/commands.json]:
```suggestion
    "clearAdvice": "You can clear the queue using `{{usage}}`.",
```

File: src/commands/music/queue.js (hunk 1 above)
Comment (davipatury) [reply src/commands/music/queue.js]: This is going to be fixed in another PR, I would have to change the CommandContext and every command that uses it.

**Comment by metehus** [comment]
you could also close #594

**Comment by davipatury** [comment]
> you could also close #594

Sorry, but this PR already has a lot of changes, so I'll leave it to another one.

**Comment by davipatury** [comment]
FUCK ME

**Comment by davipatury** [comment]
13 issues bois

File: src/locales/en-US/commands.json [hunk 9]
```diff
@@ -704,6 +727,14 @@
     "commandDescription": "Shows a random post from the pusheen tumblr.",
     "hidden": "Uwu hi there (*´∇｀)ﾉ,  yoou just got the SUPER UPER IPIE SECRETY MESSAGY °˖✧◝(⁰▿⁰)◜✧˖°\nNow that you are here.... ( ˘͈ ᵕ ˘͈♡) you could help me [here]({{link}}) (ﾉ≧ڡ≦)"
   },
+  "seek": {
+    "commandDescription": "Adjusts the volume of the music playback",
```
[the remark this answers is not in the document]
Comment (davipatury) [reply src/locales/en-US/commands.json]: oof

File: src/locales/en-US/commands.json [hunk 10]
```diff
@@ -144,7 +144,28 @@
...
+        "commandDescription": "Shuffles the queue",
+        "queueShuffled": "The queue has been shuffled!"
+      },
+      "remove": {
+        "commandDescription": "Removes a song from the queue",
+        "missingIndexParameter": "You need to give me the number of the song you want to remove from the queue",
+        "songRemoved": "{{- songName}} has been removed from the queue"
+      },
+      "jump": {
+        "commandDescription": "Removes a song from the queue",
```
Comment (LogGame-HU-DEaN) [inline src/locales/en-US/commands.json]: This does not look right either.

File: src/locales/en-US/commands.json [hunk 11]
```diff
@@ -725,6 +756,11 @@
     "commandDescription": "Shows the link to Switchblade support server",
     "clickHere": "**[Click here to join my support server](https://invite.switchblade.xyz)**"
   },
+  "bassboost": {
+    "commandDescription": "Adjusts the volume of the music playback",
```
Comment (davipatury) [inline src/locales/en-US/commands.json]: This does not look right either.

**Pull request merged**